### PR TITLE
Add node-beanstalkd-worker library types

### PR DESCRIPTION
**Pull request by SomaticIT**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

**/!\ Note and Warning: Depends on PR #39434**

### CODE REVIEW

File: types/beanstalkd-worker/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,248 @@
+// Type definitions for beanstalkd-worker 1.2
+// Project: https://github.com/burstable/node-beanstalkd-worker
+// Definitions by: Maxime LUCE <https://github.com/SomaticIT>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+import BeanstalkdClient, { BeanstalkdJobState, BeanstalkdJobStats } from "beanstalkd";
+
+export default class BeanstalkdWorker {
```
Comment (weswigham) [inline types/beanstalkd-worker/index.d.ts]: OK. So. I hate to say this, but [the package at runtime does not export a default to commonjs](https://runkit.com/embed/yij9h0iju8g3). The babel plugin (old version of babel?) they use to build hoists their `default` export into the entire module. So, I suggest:
```suggestion
export = BeanstalkdWorker;
class BeanstalkdWorker {
```
You'll need to lump all the other exported typed into a `namespace BeanstalkdWorker` to get them exported as well.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/beanstalkd-worker/index.d.ts (hunk 1 above)
Comment (SomaticIT) [reply types/beanstalkd-worker/index.d.ts]: Hello @weswigham,

I checked on the built version of the plugin and it's exported on `default` export.

```javascript
exports['default'] = BeanstalkdWorker;
module.exports = exports['default'];
```

So I think that the default export is the best way to type it.

Comment (SomaticIT) [reply types/beanstalkd-worker/index.d.ts]: No, after a runtime check, you're right, it seems to be exported badly...

Comment (SomaticIT) [reply types/beanstalkd-worker/index.d.ts]: I fixed it in commit https://github.com/DefinitelyTyped/DefinitelyTyped/pull/39435/commits/07c7d5f73cfc901df332ca496b6cf1d745a3f082

**Comment by SomaticIT** [comment]
Any news on the new version?
Thanks

File: types/beanstalkd/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,536 @@
+// Type definitions for beanstalkd 2.2
```
[the remark this answers is not in the document]
Comment (SomaticIT) [reply types/beanstalkd/index.d.ts]: Rebased

**Review by weswigham — APPROVED** [review]

**Pull request merged**